### DISCUSSION TITLE
Pre-select all albums when trying to edit some track

### DISCUSSION THREAD

**Discussion by lenrocd** [RudeySH/lastfm-bulk-edit#8] [Q&A]
I wonder why the version 0.3.0 introduced the behavior of pre-selecting all albums that contain the scrobble I am about to edit.
I find this prone to unwanted edits on albums you may not want to modify for that track (scrobble) - if you pay no attention to this pre-select all situation.
Yes, the button to easily un-select all is very welcome here, but, nonetheless, I'd prefer the old behavior: none selected when entering the edit window, and let me then select all if I really need this.
Thanks.

**Comment by RudeySH** [comment]
I don't think that was an intentional change. Give me a moment while I compare the current behavior to pre-0.3.0.

**Comment by RudeySH** [comment]
As far as I can tell, the current behavior and past behavior have always been the following:

- When editing an artist or album whose tracks were scrobbled on multiple albums, or when editing a track that was scrobbled on multiple albums, an album selection dialog appears. 

- When editing an artist or *artist track*, all albums are selected by default. By "artist track" I mean the tracks that are displayed when viewing an artist in your library. These tracks are not tied to any specific album, the scrobble count includes scrobbles across all albums, which is why all albums are selected by default during editing.

- When you're editing an album or *album track*, the album in question will be selected by default, while all other albums are unselected.  By "album track" I mean the tracks you see when you're viewing an album in your library, where the tracklist and your scrobbles are displayed.

If I'm understanding you correctly, you are saying the last bullet point is no longer the current behavior, but for me it is still working as intended. I made a recording to show the current behavior off:

![album](https://user-images.githubusercontent.com/3918757/111634707-0e35a280-87f7-11eb-84c8-ca6b124b89ac.gif)

Since v0.3.0 there's a new way to start bulk editing: viewing a track in your library will now also show an "Edit scrobbles" link on top. If the track was scrobbled across multiple albums, all albums will be selected by default. This is because a track page shows scrobbles from all albums, not just one album, so the default album selection should match your expectations.

**Reply by lenrocd** [reply]
Yes, I confirm that I have same behavior in that similar situation: in my library, for some album and I try to edit scrobbles for some track - it pre-selects only the current album where I started the edit.

Now when I try to recall past edits, I may have *not* edited some track like that - chose the edit from tracks library page directly (usually I start edit from album page in my library, as in your simulation above) and thus may have never observed that the behavior was like that on the older version as well.

Anyway, the new options added (link for edit at artist level in the library, or for the album) are very helpful.

Thanks for the very good work :-)

**Comment by hans-juergen** [comment]
I also remember it that way, i.e. the behaviour did not change compared to earlier versions of the script. But thanks to cornel for asking this directly on GitHub, so I will merge your new thread with my announcement in the Last.fm support forum.
